### ISSUE TITLE
Question: Normalizing Objective Function Outputs for MOBO

### ISSUE THREAD

**Issue by 123epsilon** [pytorch/botorch#1317]
BoTorch Version: 0.6.4

I am following this tutorial to implement multi-objective bayesian optimization: https://botorch.org/tutorials/multi_objective_bo
Essentially, I am using the exact same code, with a custom problem that attempts to balance the compress ratio of a set of files with their dice loss wrt a PyTorch image segmentation model. I negate the dice loss output so that the output lies in [-1,0] and the compression ratio output lies in [0,+inf). I am only using qNEHVI.

I have noticed some bias towards optimizing for the compression ratio. I wondered if that was a result of not standardizing the function outputs when optimizing the acquisition functions to find new candidates. In the `initialize_model()` function from the tutorial above each `SingleTaskGP` has `outcome_transform=Standardize(m=1)` which ought to normalize the output internally for the purpose of GP initialization, however, when the posterior is evaluated these outputs are evaluated on the original scale. Could this cause preferential optimization for the compress ratio since the hypervolume can be maximized more easily by scaling this objective up?

Essentially:
1) Is there a need to standardize objective outputs that differ greatly in scale when running the `optimize_acqf()` routine in the context of the above tutorial?

2) What is the correct way to implement that functionality in BoTorch?

As someone new to multiobjective optimization, I would appreciate any help.

**Comment by sdaulton** [comment]
> the compression ratio output lies in [0,+inf)

I assume compression ratio is old_size/new_size. If you instead maximized (-new_size/old_size), then the objective values of interest could be bounded by [L,0] (e.g. perhaps one would want to set L=-1 if there is no interest in parameter configurations that increase file size). This would might be helpful for setting a reference point.

There is a related discussion on weighting objectives and hypervolume improvements that may be of interest here: https://github.com/pytorch/botorch/discussions/1004. Normalizing/standardizing objectives would be equivalent to assigning a weight to each objective. 

In your case, I suspect that compression ratio is easy to model (I am guessing it is deterministic function of the parameters), whereas dice loss is probably much more difficult to model. This could lead the optimization to bias towards improving compression ratio because the model may have far greater certainty that a new design will improve compression ratio than dice loss.

**Comment by 123epsilon** [comment]
> I assume compression ratio is old_size/new_size. If you instead maximized (-new_size/old_size), then the objective values of interest could be bounded by [L,0] (e.g. perhaps one would want to set L=-1 if there is no interest in parameter configurations that increase file size). This would might be helpful for setting a reference point.

I see, previously I was setting my reference point to be -0.1 for the compress ratio, and -1.1 for the negated dice loss - strictly speaking is this wrong? Out of curiosity, is there some advantage to having bounded objective values? 


> In your case, I suspect that compression ratio is easy to model (I am guessing it is deterministic function of the parameters), whereas dice loss is probably much more difficult to model. This could lead the optimization to bias towards improving compression ratio because the model may have far greater certainty that a new design will improve compression ratio than dice loss.

Since the formulation you proposed above for the compress ratio objective would have a smaller real valued improvement for a corresponding increase in the compression ratio, is it correct to expect that this could reduce the degree to which the model favors the improvement of that objective - that is if there was no difference in certainty between the two objectives?

**Comment by sdaulton** [comment]
> Out of curiosity, is there some advantage to having bounded objective values?

Mosly, just that maximizing an objective with no upper bound (in the search space) seems a bit strange.  If the dice loss does not go to zero (and fast enough) as the compression ratio goes to infinity, then the HV of the true PF is infinite. If could model such a function, this would lead to positive utility for improvements in that direction, in the context of multi-objective BO.

> Since the formulation you proposed above for the compress ratio objective would have a smaller real valued improvement for a corresponding increase in the compression ratio, is it correct to expect that this could reduce the degree to which the model favors the improvement of that objective - that is if there was no difference in certainty between the two objectives?

I believe this question is getting at the original question, which is about how the scale of the objectives impacts EHVI, if the models are equally certain about both objectives. My previous post (https://github.com/pytorch/botorch/discussions/1004) on weighting objectives addresses this: the scale of the objectives does not have an effect on the utility of new point because HV (and HVI) is multiplicative with respect to the objectives. The effect of using the formulation I suggested above is a bit tricky to reason about with respect to the originally formulated objective since I proposed to use a negative inverse of the original objective and is not a simple rescaling.
